lib/json.c

```diff
@@ -358,7 +357,7 @@ sr_json_parse_ex(struct sr_json_settings *settings,
                 case ']':
                     if (top && top->type == SR_JSON_ARRAY)
                         flags = (flags & ~ (flag_need_comma | flag_seek_value)) | flag_next;
-                    else if (!top || (!state.settings.settings & SR_JSON_RELAXED_COMMAS))
+                    else if (!top || !(state.settings.settings & SR_JSON_RELAXED_COMMAS))
```

Comment:
Same as the last time - have you verified that this is correct? The commit message is a tad light.

Reply:
Yes, under the (currently valid) assumption that `SR_JSON_RELAXED_COMMAS` is the only possible flag in `state.settings.settings`, these expressions are equivalent.

Moreover, the latter appears to be what was intended by the authors. In the original statement, it seems outlandish to _logically_ negate a bitfield and then check _bitwise_ for the presence of a flag. Either the negation should have been bitwise or the proposed parenthesization should be used. Both of these solutions are, naturally, equivalent.

Reply:
Right, so I looked at it some more and it makes even less sense to me. AFAICT, https://github.com/abrt/satyr/blob/8b5547b89b712b39a59f1d8b366e7de0f5f46108/lib/json.c#L642 is the only place where the settings are initialized in some way. Since it’s zeroed out, negating the field would give us 1, which is also the value of `SR_JSON_RELAXED_COMMAS`. Negating the expression as a whole gives us an equivalent result, but just reads terribly wrong.

Reply:
So I’ll let it slide, so to speak, with the proviso that we update the code eventually.

Reply:
Also, crazy shit like this makes me wonder why we don’t just use an external shared library.